testing/wireguard-tools/APKBUILD

```diff
@@ -0,0 +1,65 @@
+# Contributor: Stuart Cardall <developer@it-offshore.co.uk>
+# Maintainer: Stuart Cardall <developer@it-offshore.co.uk>
+
+pkgname=wireguard-tools
+pkgver=0.0.20170223
+pkgrel=0
+pkgdesc="Next generation secure network tunnel: userspace tools"
+arch='all'
+url='https://www.wireguard.io'
+license="GPLv2"
+makedepends="libmnl-dev sparse"
+subpackages="$pkgname-doc $pkgname-bash-completion:bashcomp:noarch \
+		$pkgname-tests:tests:noarch"
+source="https://git.zx2c4.com/WireGuard/snapshot/WireGuard-$pkgver.tar.xz"
+builddir="$srcdir"/WireGuard-$pkgver
+
+build() {
+	cd "$builddir"
+	make -C src/tools || return 1
+}
+
+package() {
+	cd "$builddir"
+
+	install_if="linux-${_flavor}=${_kver} $pkgname"
+	mkdir -p "$pkgdir"/usr/share/doc/$_name
+
+	make -C src/tools \
+		DESTDIR="$pkgdir" \
+		WITH_BASHCOMPLETION=yes \
+		WITH_WGQUICK=yes \
+		WITH_SYSTEMDUNITS=no \
+		install || return 1
+
+	find "$builddir"/contrib/examples -name '.gitignore' -delete
+	cp -rf "$builddir"/contrib/examples "$pkgdir"/usr/share/doc/$_name/
+}
+
+bashcomp() {
+	depends="bash"
+	pkgdesc="Wireguard BASH completions"
+
+	mkdir -p "$subpkgdir"/usr
+	mv "$pkgdir"/usr/share "$subpkgdir"/usr
+}
+
+tests() {
+	pkgdesc="Wireguard test suites"
+
+	mkdir -p "$subpkgdir"/usr/share/wireguard
+	mv "$builddir"/src/tests "$subpkgdir"/usr/share/wireguard
+	cp -rf "$builddir"/contrib/external-tests "$subpkgdir"/usr/share/wireguard
+}
+
+check() {
+	return 0
```

Comment:
I'm not sure how this would work for you, but the WireGuard module actually has built-in selftests when built in DEBUG mode. When enabled, either the module inserts and everything is fine, or it fails to insert, which implies the tests have failed (and you get something in dmesg). This probably is too difficult to do in a package though...